File: Pset2_-_Arrays/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    /* ask user for a string */
    string s = get_string("Text: ");

    /* initialize variables */
    int n = strlen(s);
    float countL = 0;
    int i = 0;
    int c;
    float countW = 0;
    float countS = 0;

    /* scans length of string */
    while ((c = s[i++]) && (i <= n))
    {
        /* counts letters in string */
        if ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'))
        {
            countL++;
        }

        /* counts words in string */
        if (c == ' ' && c + 1 != ' ')
        {
            countW++;
        }

        /* counts sentences in string */
        if (c == '.' || c == '?' || c == '!')
        {
            countS++;
        }

    }

    /* counts the last word in string */
    if (s[0])
    {
        countW++;
    }

    /* calculate variables part of equation */
    float L = (countL / countW) * 100;
    float S = (countS / countW) * 100;
    float gradeLevel = 0.0588 * L - 0.296 * S - 15.8;

    /* print results */
    if (gradeLevel < 0)
    {
        printf("Before Grade 1\n");
    }

    if (gradeLevel > 16)
    {
        printf("Grade 16+\n");
    }

    if (gradeLevel > 0 && gradeLevel < 16)
    {
        printf("Grade %1.0f\n", gradeLevel);
    }

}

File: Pset6_-_Python/readability.py
from cs50 import get_string

# ask user for a string
s = get_string("Text: ");

# initialize variables
n = len(s)
countL = 0
i = 0
countW = 0
countS = 0
c = 0

# scans length of string
for i in range(len(s)):
    # counts letters in string
    if (s[i] >= 'A' and s[i] <= 'Z') or (s[i] >= 'a' and s[i] <= 'z'):
        countL += 1

    # counts words in string
    if s[i] == ' ' and s[i+1] != ' ':
        countW += 1

    # counts sentences in string
    if s[i] == '.' or s[i] == '?' or s[i] == '!':
        countS += 1

# counts the last word in string
if s[0]:
    countW += 1

# calculate variables part of equation
L = (countL / countW) * 100
S = (countS / countW) * 100
gradeLevel = 0.0588 * L - 0.296 * S - 15.8

# print results
if gradeLevel < 0:
    print("Before Grade 1")

if gradeLevel > 16:
    print("Grade 16+")

if gradeLevel > 0 and gradeLevel < 16:
    print(f"Grade {round(gradeLevel)}")